com.unity.cinemachine/Editor/Upgrader/CinemachineUpgradeManager.cs

```diff
@@ -172,217 +153,36 @@ struct UniqueExposedReference
-                    }
-                }
-            }
-            
-            m_CurrentSceneOrPrefab = string.Empty;
-        }
-
-        /// <summary>
-        /// Opens every scene in the project and upgrades:
-        /// 1. prefab instances using Conversion link,
-        /// 2. non-prefabs,
-        /// 3. cleans obsolete components,
-        /// 4. updates animation references.
-        /// </summary>
-        /// <param name="conversionLinksPerScene">A per scene (scene index) list of links between prefab instances
-        /// and their upgraded copies that contain the original prefab modifications.</param>
-        void UpgradeAllScenes(Dictionary<int, List<ConversionLink>> conversionLinksPerScene)
-        {
-            for (var s = 0; s < m_SceneManager.SceneCount; ++s)
-            {
-                // 0. Open scene
-                var scene = OpenScene(s);
-                var rootObjects = scene.GetRootGameObjects();
-                var timelineManager = new TimelineManager(scene);
-                var upgradables = GetUpgradables(rootObjects, m_ObjectUpgrader.RootUpgradeComponentTypes, true);
-                var upgradedObjects = new HashSet<GameObject>();
-                
-                // 1. Prefab instances
-                UpgradePrefabInstances(upgradedObjects, conversionLinksPerScene[s], timelineManager);
-
-                // 2. Non-prefabs
-                UpgradeNonPrefabs(upgradables, upgradedObjects, timelineManager);
-                
-                // 3. Fix up object references
-                UpgradeObjectReferences(rootObjects);
-
-                // 4. Clean up obsolete components
-                foreach (var go in upgradedObjects)
-                    m_ObjectUpgrader.DeleteObsoleteComponents(go);
-
-                // 4. AnimationReferences
-#if CINEMACHINE_TIMELINE
-                UpdateAnimationReferences(TimelineManager.GetPlayableDirectors(scene));
-#endif
-
-                // 5. Save scene
-                EditorSceneManager.MarkSceneDirty(scene);
-                EditorSceneManager.SaveScene(scene);
-            }
-        }
-
-       
         /// <summary>
         /// First, restores modifications in all prefab instances in the current scene by copying data from the linked
         /// converted copy of the prefab instance.
         /// Then, restores timeline references to any of these upgraded instances.
         /// </summary>
         /// <param name="conversionLinks">Conversion links for the current scene</param>
         /// <param name="timelineManager">Timeline manager for the current scene</param>
-        void UpgradePrefabInstances(
-            HashSet<GameObject> upgradedObjects, 
-            List<ConversionLink> conversionLinks, TimelineManager timelineManager)
+        /// <param name="upgradeReferencables">
+        /// True, then only referencable prefab instances are upgraded.
+        /// False, then only non-referencable prefab instances are upgraded.</param>
```

Comment:
What if a prefab has both a referencable and a non-referencable? e.g. dolly vcam + path in the same prefab. I think you have to revisit the logic in this function to handle those cases correctly.